lib/internal/main/test_runner.js

```diff
@@ -5,18 +5,23 @@ const {
 } = primordials;
 
 const {
-  prepareMainThreadExecution,
   markBootstrapComplete,
+  prepareTestRunnerMainExecution,
 } = require('internal/process/pre_execution');
 const { isUsingInspector } = require('internal/util/inspector');
 const { run } = require('internal/test_runner/runner');
 const { parseCommandLine } = require('internal/test_runner/utils');
+const { getOptionValue } = require('internal/options');
 const { exitCodes: { kGenericUserError } } = internalBinding('errors');
 let debug = require('internal/util/debuglog').debuglog('test_runner', (fn) => {
   debug = fn;
 });
 
-prepareMainThreadExecution(false);
+const isTestIsolationNone = getOptionValue('--test-isolation') === 'none';
+
+// We set initializeModules to false as we want to load user modules in the test runner run function
+// if we are running with --test-isolation=none
+prepareTestRunnerMainExecution(!isTestIsolationNone);
```

Comment:
very nit: I think saying "disabled" will be more broadly/immediately understood than "none".

```suggestion
const isTestIsolationDisabled = getOptionValue('--test-isolation') === 'none';

// We set initializeModules to false as we want to load user modules in the test runner run function
// if we are running with --test-isolation=none
prepareTestRunnerMainExecution(!isTestIsolationDisabled);
```

Reply:
applied the suggestion!